### DISCUSSION TITLE
How to handle fetch errors

### DISCUSSION THREAD

**Discussion by idahogurl** [vercel/swr#809] [Q&A]
I've looked through the documentation and I cannot find how to handle exceptions  thrown in my `fetcher` function. This is what I came up with.

```
const [fetchError, setFetchError] = useState();
const fetcher = (url, params) => {
    return Axios.post(url, {
      headers: { Authorization: `Bearer ${params.accessToken}` },
    }).then((res) => res).catch(e => { setFetchError(e) });
  };

const { data, error } = useSWR([path, JSON.stringify({ search })], fetcher);

return {
    data: data,
    loading: !fetchError && !error && !data,
    error: fetchError || error,
  };
```

Shouldn't `useSWR` handle those?

**Comment by mimoid-prog** [comment]
https://swr.vercel.app/docs/error-handling

> If an error is thrown inside fetcher, it will be returned as error by the hook.

Isn't that what you looking for?
